jane/doc/extensions/stack/reference.md

```diff
@@ -121,51 +126,47 @@ type-checker ensures that it does not outlive the region that it lives in,
 which is the entire body of `f`.
 
 ### Other regions
+
 As well as function bodies, a region is also placed around:
 
   - Loop bodies (`while` and `for`)
   - Lazy expressions (`lazy ...`)
-  - Module bindings (`let x = ...` at module level, including in sub-modules)
+  - Module bindings (`let x = ...;;` at module level, including in sub-modules)
 
 Module bindings are wrapped in regions to enforce the rule (as mentioned above)
 that modules never contain local values.
 
-Additionally, it is possible to write functions that do *not* have a region
-around their body, which is useful to write functions that return
-stack-allocated values. See "Use exclave_ to return a local value" below.
+Additionally, it is possible to write functions whose region ends before the
+function does, which is useful to write functions that return
+stack-allocated values. See "Use `exclave_` to return a local value" below.
 
 ### Nested regions
-Regions may nest, for instance when one function calls another. Stack-allocated
-values always lives in the surrounding region, since at runtime one can
-only allocate in the current stack frame. There always exists a region at the
-outermost. This region never ends and is where global (such as heap-allocated)
-values live.
 
-To better understand the nested structure, imagine we are in a region, and the
-local variables in scope are further subdivided into two cases:
-
-  - **Outer-region local**: lives in an outer region, not the current region.
+Regions nest, for instance when defining a local function. Stack-allocated
+values always live in the inner-most region, since at runtime one can
+allocate only in the current stack frame. There is an outermost region,
+including an entire file. This region never ends and is where global (heap-allocated)
+values live.
 
-  - **Any-region local**: lives in an unknown region, potentially the current
-    region.
+One subtlety is that we wish to treat `local` variables from the current region
+differently than `local` variables from an enclosing region. Variables from an
+enclosing region are called *regional*.
 
 For instance:
 
 ```ocaml
 let f () =
-  let stack_ outer = ref 42 in
+  let local_ outer = ref 42 in
   let g () =
-    let stack_ inner = ref 42 in
+    let local_ inner = ref 42 in
     ??
   in
   ...
 ```
 
 At the point marked `??` inside `g`, both `outer` and `inner` are
-stack-allocated values and thus local. However, only `inner` is any-region
-local, having been allocated in `g`'s region. The value `outer` is instead
-outer-region local: it is stack-allocated but from a region outer than `g`'s
-own.
+local values; they will both be allocated on the stack at runtime. However,
```

Comment:
I think we want to reduce the impression that "local values goes to stack"; here we can strictly stick to locality and escaping.

Reply:
OK. Are you suggesting to drop the point about stack allocation? I’m fine with that.

Reply:
Please just go ahead and drop that bit of sentence to resolve this. Thanks